NCS.DSS.Customer/AzureSearchDataSyncTrigger/CustomerSearchDataSyncTrigger.cs

```diff
@@ -1,91 +1,90 @@
-
-                var batch = IndexDocumentsBatch.MergeOrUpload(customers);
+            { 
+                try
+                {      
+                    var customers = new List<Models.CustomerSearch>();
+                    foreach (var doc in documents)
+                    {
+                        var custJson = JsonSerializer.Serialize(doc);
+                        var customer = JsonSerializer.Deserialize<Models.CustomerSearch>(custJson);
+                        if(doc.CustomerId == null && doc.id != null)
+                            customer.CustomerId = doc.id;
+                        else if(doc.id == null && doc.CustomerId != null)
+                            customer.CustomerId = doc.CustomerId;
+
+                        customers.Add(customer);
+                    }
+                    var custFiltered = customers.Where(d => d.CustomerId != null);
+                    if (custFiltered.Any())
+                    {
+                        var batch = IndexDocumentsBatch.MergeOrUpload(custFiltered);
 
+                        _logger.LogInformation("attempting to merge docs to azure search");
 
-                try
-                {
-                    _log.LogInformation("attempting to merge docs to azure search");
+                        _logger.LogInformation("Document IDs : {Ids}", string.Join(',', custFiltered.Select(d => d.CustomerId).ToArray()));
 
-                    var results = await client.IndexDocumentsAsync(batch);
 
-                    var failed = results.Value.Results.Where(r => !r.Succeeded).Select(r => r.Key).ToList();
+                        var results = await client.IndexDocumentsAsync(batch);
 
-                    if (failed.Count > 0)
-                    {
-                        _loggerHelper.LogInformationMessage(_log, correlationId, string.Format("Failed to index some of the documents: {0}", string.Join(", ", failed)));
-                    }
+                        var failed = results.Value.Results.Where(r => !r.Succeeded).Select(r => r.Key).ToList();
 
-                    _log.LogInformation("successfully merged docs to azure search");
+                        if (failed.Count > 0)
+                        {
+                            _logger.LogInformation("Failed to index some of the documents: {errors}", string.Join(", ", failed));
+                        }
 
+                        _logger.LogInformation("successfully merged docs to azure search");
+                    }
+                    var custFailed = customers.Where(d => d.CustomerId == null);
+                    if (custFailed.Any())
+                    {
+                        _logger.LogInformation("Below list of documents can't be processed as they are missing with Document Key");
+                        foreach (var doc in custFailed.Where(d => d.CustomerId == null))
+                        {
+                            _logger.LogInformation(JsonSerializer.Serialize(doc));
```

Comment:
Change this to be a log like "{doc} is missing a document key", or something to that affect which explains why this customer failed